### PR TITLE
Implementing the UploadSnapshotsController

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #991

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
The `browser-tools` Orb is failing with the following error:

```bash
GECKODRIVER_VERSION_NUM="$(echo "$GECKODRIVER_VERSION" | sed -E 's/v//')"

if geckodriver --version | grep "$GECKODRIVER_VERSION_NUM" >/dev/null 2>&1; then
  echo "$(geckodriver --version) has been installed to $(command -v geckodriver)"
else
  echo "Something went wrong; the specified version of Geckodriver could not be installed"
  exit 1
fi

Selected version of Geckodriver is: v0.32.2
Geckodriver is not currently installed; installing it
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.haxx.se/docs/sslcerts.html
```

**Review by carolyncole — CHANGES_REQUESTED** [review]
I feel like there will be more than one file on a work, but it seems like the controller method is dealing with a singular file.

I may have missed the point, os let me know if I did...

File: app/controllers/upload_snapshots_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+class UploadSnapshotsController < ApplicationController
+  # POST /upload-snapshots/:work_id
+  def create
+    @upload_snapshot = upload.create_snapshot
```
[the remark this answers is not in the document]
Comment (jrgriffiniii) [reply app/controllers/upload_snapshots_controller.rb:7]: I am very sorry, I should have included a UI View for this (I am starting this right at the moment). Essentially, `Works` have multiple `uploads` (S3 files), which then have multiple `UploadSnapshots`. There is a one-to-many relationship between an `upload` and `UploadSnapshots`, however, I use the `uri` attribute to resolve to a unique `UploadSnapshot`.

I can be certain that the UI View clarifies this.

File: db/migrate/20230303165757_create_upload_snapshots.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 class CreateUploadSnapshots < ActiveRecord::Migration[6.1]
   def change
     create_table :upload_snapshots do |t|
-      t.string :uri, null: false
+      t.string :filename
+      t.string :url
+      t.bigint :version
```
Comment (carolyncole) [inline db/migrate/20230303165757_create_upload_snapshots.rb:6]: Carolyn's imagination....
For each work you would have many entires:
{ files:[
   {filename:  'name1', checksum: 'abc1232'},
   {filename:  'name2', checksum: 'abc1232'},
   {filename:  'name3', checksum: 'abc1232'},
]
}

**Comment by jrgriffiniii** [comment]
This is deprecated in favor of an improved implementation which I shall issue as a new pull request.

**Pull request closed without merging**